core/src/main/java/com/adyen/checkout/core/components/CheckoutController.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2025 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 15/7/2025.
+ */
+
+package com.adyen.checkout.core.components
+
+import android.content.Intent
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import com.adyen.checkout.core.action.data.Action
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+
+// TODO - KDocs
+class CheckoutController {
+
+    private val _events = MutableSharedFlow<Event>(
+        replay = 0,
+        extraBufferCapacity = 1,
+        onBufferOverflow = BufferOverflow.DROP_OLDEST,
+    )
+    internal val events: SharedFlow<Event> = _events.asSharedFlow()
+
+    fun submit() {
+        _events.tryEmit(Event.Submit)
+    }
+
+    fun handleAction(action: Action) {
```

Comment:
I don't think we need handle action here, since we will be getting that as a return type from the merchants in the callback.

Reply:
What if a merchant just wants to use our component to handle actions? This would give the possibility to do that, but maybe we could think about other solutions as well

Reply:
I suggest we discuss that. For now you can leave a `TODO` here for that maybe?